### ISSUE TITLE
Provide user-friendly default behavior when processing invalid samples through Brain methods

### ISSUE THREAD

**Issue by brimoor** [voxel51/fiftyone-brain#30]
**Background**

All brain methods currently have a `validate` flag, set to `False` by default, that control whether validation is performed on samples when deciding if valid data was provided or not.

For example, validation might complain if a user requests an operation that requires logits for a prediction, but none are found. Or, validation might enforce certain constraints on input images such as grayscale vs color.

**Objective**

Provide the most user-friendly experience possible when users invoke brain methods, allowing them to:
1. not worry unnecessarily about the format of their data
2. rest assured that egregiously unnecessary computation is not being performed without their knowledge
3. small errors do not result in catastrophic errors
